Add new rule on files outside of meta/, plugins/, roles/, and playbooks/ (#199)
ansible-collections/overview: collection_requirements.rst

  The core team (which maintains ansible-core) has committed not to use these directories for
  anything which would conflict with the uses we've specified.

Other directories
-----------------

Collections MUST not use files outside ``meta/``, ``plugins/``, ``roles/`` and ``playbooks/`` in public plugins, roles and playbooks they contain. They must work if every file or directory is deleted from the installed collection except these directories and their contents.

> In the previous paragraph, `"public"` means that internal plugins, roles and playbooks, for example which are part of tests, only used to release the collection, etc. are not affected. Everything that can be used from other collections or user playbooks and roles, as well as playbooks that can be called by FQCN, is public.

Author: How about the following instead:
```suggestion
In the previous paragraph, `"public"` means that internal plugins, roles and playbooks are not affected. Internal means that they are part of tests, only used to release the collection, etc. Everything that can be used from other collections or user playbooks and roles, as well as playbooks that can be called by FQCN, is public.
```

> In the previous paragraph, `"public"` means that internal plugins, roles and playbooks, for example which are part of tests, only used to release the collection, etc. are not affected. Everything that can be used from other collections or user playbooks and roles, as well as playbooks that can be called by FQCN, is public.

Reviewer: more sentences is good for readability and understanding

> Collections MUST not use files outside ``meta/``, ``plugins/``, ``roles/`` and ``playbooks/`` in public plugins, roles and playbooks they contain. A collection must work if every file or directory is deleted from the installed collection except these directories and their contents.
> 
> In the previous paragraph, `"public"` means that internal plugins, roles and playbooks are not affected. Internal means that they are part of tests, only used to release the collection, etc. Everything that can be used from other collections or user playbooks and roles, as well as playbooks that can be called by FQCN, is public.

Reviewer 2: ```suggestion
Collections MUST not use files outside ``meta/``, ``plugins/``, ``roles/`` and ``playbooks/`` in any plugin, role, or playbook that can be called by FQCN, used from other collections, or used from user playbooks and roles.  A collection must work if every file or directory is deleted from the installed collection except these directories and their contents.

Internal plugins, roles and playbooks (artifacts used only in testing, or only to release the collection, or only for some other internal purpose and not used externally) are exempt from this rule and may rely on files in other directories.
```
